scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -1805,13 +1805,21 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
   /* ------------- TYPES ---------------------------------------------------- */
 
   def typedOpt(): Option[Type] =
-    if (token.is[Colon]) {
+    if (token.is[Colon] || token.is[ColonEol]) {
```

Comment:
This is really a workaround since we can possibly only see if colon should be ColonEol during parsing, but we drop all white spaces before that, so we need to guess and then fix it here. This will add additional IdentTokens, but it should be ok still.